### PR TITLE
newtype: minor improvements from code review

### PR DESCRIPTION
**Pull request by kodebach**
## Basics

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [x] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

- [x] I added unit tests
- [x] I ran all tests locally and everything went fine
- [x] affected documentation is fixed
- [x] I added code comments, logging, and assertions (see doc/CODING.md)
- [x] meta data is updated (e.g. README.md of plugins)

## Review

@markus2330 please review, this continues from #2505

### CODE REVIEW

**Comment by kodebach** [comment]
@markus2330 please review

This is `ready to merge` the failed build job is duplicate and was successfully repeated

File: src/error/specification [hunk 1]
```diff
@@ -310,10 +310,11 @@ ingroup:plugin
 module:network
 
 number:52
-description:could not type check value of key
+description:error in the type plugin
```
Comment (markus2330) [inline src/error/specification]: Isn't "error in the type plugin" way too generic? What about "the value does not match with the type of the key"?

Comment (kodebach) [reply src/error/specification]: The description is going to be removed with the error concept anyways (hopefully soon). The reasons used in code are much more descriptive.

Comment (markus2330) [reply src/error/specification]: As long as we have the old error system we need to live with what we have. Hopefully @Piankero finishes soon!

**Pull request merged**